app/src/main/java/org/asteroidos/sync/connectivity/IConnectivityService.java

```diff
@@ -0,0 +1,39 @@
+package org.asteroidos.sync.connectivity;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * A connectivity service is a module that can exchange data with the watch. It has to implement {@link IService}
+ * and additional functions regarding connectivity from {@link IConnectivityService}.
+ */
+public interface IConnectivityService extends IService {
+
+    enum Direction{
+        RX,
+        TX
+    }
+
+    public HashMap<UUID, Direction> getCharacteristicUUIDs();
+
+    public UUID getServiceUUID();
+
+    /**
+     * <code>onReceive()</code> is called when data from the watch is received.
+     *
+     * <pre>
+     * &#064;Override
+     * <b>public void</b> onReceive(UUID uuid, byte[] data) {
```

Comment:
are these autogenerated ? I don't think they help.
the reader should be able to copy paste the prototype and add `@Override` without us holding their hand.
Also these are difficult to maintain, look the interface is already wrong, it says public void but the real interface is public Boolean